pwndbg/commands/pie.py

```diff
@@ -27,11 +27,21 @@ def get_exe_name():
     """
     path = pwndbg.auxv.get().get('AT_EXECFN')
 
-    if path is not None:
+    # When GDB is launched on a file that is a symlink to the target,
+    # the AUXV's AT_EXECFN stores the absolute path of to the symlink.
+    # On the other hand, the vmmap, if taken from /proc/pid/maps will contain
+    # the absolute and real path of the binary (after symlinks).
+    # And so we have to read this path here.
+    real_path = pwndbg.file.readlink(path)
```

Comment:
We might also just be able to read `/proc/self/exe`, versus using AT_EXECFN.  I think the only reason we don't do that is for QEMU, since it can't emulate what we see.

Perhaps we should check for QEMU first, and just use `/proc/self/exe` otherwise?